onc-healthit/lantern-back-end
# Db name change

The current setup creates the default database named `postgres` and the default admin user called `postgres`. this PR updates the default to user `lantern` for both values instead.

---

To check this, you will need to remove the docker database volume on your machine so the database can be rebuilt with the new settings. If you are using docker-compose, you can do this with:

```
docker-compose down -v
```

When you restart using docker-compose, there may be issues with the prometheus-postgres-adapter connecting to the postgres database for timing issues. If this is the case, run `docker-compose down` **with no other arguments**, and then run `docker-compose up` again. We will address the timing issue in a later PR.

If you are doing this on a container-by-container basis, you will need to stop the container (e.g., `pg_prometheus`) and remove the particular volume (e.g., `pgdata`) using:

```
docker stop pg_prometheus
docker rm pg_prometheus
docker volume rm pgdata
```


**To Test**

*Make sure to copy env.sample to .env*

Creating docker containers individually

* After removing the volumes as stated above, create the docker container for the db using the instructions in the README.
* Try to connect to the database using the following command:
    ```
    docker exec -it pg_prometheus psql -U postgres
    ```
    This should **fail**
* Try to connect to the database using the following command:
    ```
    docker exec -it pg_prometheus psql -U lantern
    ```
    This should **succeed**
* List the existing databases in the psql terminal using `\l`.

    You should see a database named `lantern` listed.

Using docker-compose

* Remove the volumes again using the instructions above.

* Restart the containers using `docker-compose` as described in the README (noting you may need to do it once, have it fail, and do it again as described above).

* Try to connect to the database using the following command:
    ```
    docker exec -it pg_prometheus psql -U postgres
    ```
    This should **fail**
* Try to connect to the database using the following command:
    ```
    docker exec -it pg_prometheus psql -U lantern
    ```
    This should **succeed**
* List the existing databases in the psql terminal using `\l`.

    You should see a database named `lantern` listed.
----

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**

Waiting on https://github.com/onc-healthit/lantern-back-end/pull/20 to update vendor files

  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-66
  - vendor/ directories have been updated
    - New dependencies have been added
    - Local dependencies have been updated
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - vendor/ directories have been updated
    - New dependencies have been added
    - Local dependencies have been updated
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge is complete.
